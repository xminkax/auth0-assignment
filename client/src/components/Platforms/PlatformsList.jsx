import React from 'react';
import PropTypes from 'prop-types';
import PlatformItem from './PlatformItem';

const PlatformsList = ({ platforms }) => (
  <div className="platforms-list-wrapper">
    {
      !platforms || platforms.length === 0 ?
        <div className="no-records"><h2>No records</h2></div> :
        <ul className="list">
          {
            platforms.map(platform => (
              <PlatformItem
                key={platform.id}
                platform={platform}
              />))
          }
        </ul>
    }
  </div>
);

export default PlatformsList;

PlatformsList.propTypes = {
  platforms: PropTypes.array.isRequired, // eslint-disable-line react/forbid-prop-types
};
