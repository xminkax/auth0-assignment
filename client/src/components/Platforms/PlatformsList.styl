.platforms-list-wrapper
  max-width 1016px
  margin 0 auto
  padding 0 $padding-base-horizontal
  @media (min-width 992px)
    width 970px

  .no-records
    display flex
    justify-content center

    h2
      +breakpoint("tablet", "max")
        font-size $font-size-h3

  .list
    list-style none
    display flex
    flex-wrap wrap
    padding 0
