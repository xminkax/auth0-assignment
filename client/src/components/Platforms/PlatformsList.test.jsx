import React from 'react';
import renderer from 'react-test-renderer';
import PlatformsList from './PlatformsList';

it('renders platforms list', () => {
  const data = [{ title: 'Android', imageName: 'android', id: '6ymHMrgkH6q6WGggqQ2aMs' }, {
    title: 'Angular 2+',
    imageName: 'angular',
    id: '5ZDQx5dW6cqweY26GqK8s6',
  }, { title: 'AngularJs', imageName: 'angular', id: '662lRdvU6AKOG6AkQKY2eU' }, {
    title: 'Django',
    imageName: 'python',
    id: '5cAqCgRaukEIIo2g4Cs6Oi',
  }];
  const component = renderer.create(
    <PlatformsList platforms={data} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
