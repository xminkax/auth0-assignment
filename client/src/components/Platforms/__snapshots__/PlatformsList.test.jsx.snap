// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders platforms list 1`] = `
<div
  className="platforms-list-wrapper"
>
  <ul
    className="list"
  >
    <li
      className="platforms-item"
    >
      <div
        className="wrapper"
      >
        <div
          className="asset-platform"
        >
          <img
            alt="android"
            className="image"
            src="../../assets/android.png"
          />
        </div>
        <div
          className="title"
        >
          Android
        </div>
        <div
          className="detail-title"
        >
          Authenticate your 
          Android
           app
        </div>
        <div
          className="connections"
        >
          <a
            className="btn btn-lg btn-success"
            href="/authenticate/react/bankid/"
          >
            Show connections
          </a>
        </div>
      </div>
    </li>
    <li
      className="platforms-item"
    >
      <div
        className="wrapper"
      >
        <div
          className="asset-platform"
        >
          <img
            alt="angular"
            className="image"
            src="../../assets/angular.png"
          />
        </div>
        <div
          className="title"
        >
          Angular 2+
        </div>
        <div
          className="detail-title"
        >
          Authenticate your 
          Angular 2+
           app
        </div>
        <div
          className="connections"
        >
          <a
            className="btn btn-lg btn-success"
            href="/authenticate/react/bankid/"
          >
            Show connections
          </a>
        </div>
      </div>
    </li>
    <li
      className="platforms-item"
    >
      <div
        className="wrapper"
      >
        <div
          className="asset-platform"
        >
          <img
            alt="angular"
            className="image"
            src="../../assets/angular.png"
          />
        </div>
        <div
          className="title"
        >
          AngularJs
        </div>
        <div
          className="detail-title"
        >
          Authenticate your 
          AngularJs
           app
        </div>
        <div
          className="connections"
        >
          <a
            className="btn btn-lg btn-success"
            href="/authenticate/react/bankid/"
          >
            Show connections
          </a>
        </div>
      </div>
    </li>
    <li
      className="platforms-item"
    >
      <div
        className="wrapper"
      >
        <div
          className="asset-platform"
        >
          <img
            alt="python"
            className="image"
            src="../../assets/python.png"
          />
        </div>
        <div
          className="title"
        >
          Django
        </div>
        <div
          className="detail-title"
        >
          Authenticate your 
          Django
           app
        </div>
        <div
          className="connections"
        >
          <a
            className="btn btn-lg btn-success"
            href="/authenticate/react/bankid/"
          >
            Show connections
          </a>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
